posthog/models/test/test_integration_model.py

```diff
@@ -231,3 +231,88 @@ def test_refresh_access_token_handles_errors(self, mock_post, mock_reload):
         assert integration.errors == "TOKEN_REFRESH_FAILED"
 
         mock_reload.assert_not_called()
+
+
+class TestGoogleCloudIntegrationModel(BaseTest):
+    mock_keyfile = {
+        "type": "service_account",
+        "project_id": "posthog-616",
+        "private_key_id": "df3e129a722a865cc3539b4e69507bad",
+        "private_key": "-----BEGIN PRIVATE KEY-----\nTHISISTHEKEY==\n-----END PRIVATE KEY-----\n",
+        "client_email": "hog-pubsub-test@posthog-301601.iam.gserviceaccount.com",
+        "client_id": "11223344556677889900",
+        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+        "token_uri": "https://oauth2.googleapis.com/token",
+        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+        "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/not-a-topic%40posthog-616.iam.gserviceaccount.com",
+        "universe_domain": "googleapis.com",
```

Comment:
Shame we couldn't do this generically... Feels like a "standard" oauth flow is in there somewhere which we could simulate instead of using their custom SDK to do it but I'm also not gonna push too hard for it. Maybe if we end up with another case in the future we can try and implement the standard service token oauth refresh

Reply:
Their keyfile is pretty specific to Google, and this isn't a normal OAuth flow, but just a "service account refresh" flow, which _can_ be considered kid of OAuth if you squint. I'm not too sure actually which part is very Google-specific and which is universal. In any case we'll need a large "if" block for a lot of google-specific stuff.

In the future if we need to add some non-Google integration that does the same, we can rename `GoogleCloudIntegration` to `ServiceAccountIntegration` and refactor accordingly.